spline-shadow/persistence/src/main/scala/za/co/absa/spline/persistence/model/persistenceModel.scala

```diff
@@ -17,66 +16,116 @@
 
 package za.co.absa.spline.persistence.model
 
-case class Progress(timestamp: Long, readCount: Long, _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
+case class Progress(timestamp: Long, readCount: Long,
+  _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
 
-case class Execution(id: String, dataTypes: Seq[DataType], startTime: Option[Long], endTime: Option[Long], extra: Map[String, Any], _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
+case class Execution(id: String, dataTypes: Seq[DataType], startTime: Option[Long], endTime: Option[Long],
+  extra: Map[String, Any], _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None)
+  extends DocumentOption
```

Comment:
What is `DocumentOption`?

Reply:
It is similarly to PolymorphicDocumentOption an trait provided by Scarango and was as well added into our code in serialization PR: https://github.com/AbsaOSS/spline/pull/130/files .
There is another one called "Edge". 

In Scarango these were also used to require essential params to be defined on all documents and all edges correspondingly. I think it would be useful for that same purpose for us even now. So I will add those into them.